src/arizona_template_compiler.erl

```diff
@@ -424,7 +466,7 @@ changeable_vars_2([], _, T) ->
     changeable_vars_1(T).
 
 norm_changeable_vars(BlockId, Vars) ->
-    [{K, Id -- BlockId} || {K, Id} <- Vars].
+    [{K, Id -- BlockId} || {K, Id} <- Vars, Id -- BlockId =/= []].
```

Comment:
It'd probably be easier to read if there were some parens, e.g. `(Id -- BlockId) =/= []`.